gtsfm/runner/gtsfm_runner_base.py

```diff
@@ -321,13 +329,28 @@ def run(self) -> GtsfmData:
 
         with performance_report(filename="correspondence-generator-dask-report.html"):
             correspondence_generation_start_time = time.time()
+            keypoints_from_all_corr_generators: List[List[Keypoints]] = []
+            putatutive_corr_idxs_from_all_corr_generators: List[Dict[Tuple[int, int], np.ndarray]] = []
+
+            for corr_generator in self.scene_optimizer.correspondence_generators:
+                (
+                    keypoints_list,
+                    putative_corr_idxs_dict,
+                ) = corr_generator.generate_correspondences(
+                    client,
+                    self.loader.get_all_images_as_futures(client),
+                    image_pair_indices,
+                )
+
+                keypoints_from_all_corr_generators.append(keypoints_list)
+                putatutive_corr_idxs_from_all_corr_generators.append(putative_corr_idxs_dict)
+
             (
                 keypoints_list,
                 putative_corr_idxs_dict,
-            ) = self.scene_optimizer.correspondence_generator.generate_correspondences(
-                client,
-                self.loader.get_all_images_as_futures(client),
-                image_pair_indices,
+            ) = correspondence_utils.merge_correspondences(
+                keypoints_from_all_corr_generators,
+                putatutive_corr_idxs_from_all_corr_generators,
```

Comment:
What if there's only one generator?

Reply:
Could we avoid calling this unless generators > 1?

Reply:
Good point.

Reply:
Actually, the function will be a no-op and just return early if we have just 1 correspondence generator. Leaving it as is.